<launch>
  <!-- MAVROS APM Launch with Distance Sensor Plugin -->
  <include file="$(find mavros)/launch/node.launch">
    <arg name="fcu_url" value="/dev/ttyTHS1:57600"/>
    <arg name="gcs_url" value=""/>
    <arg name="tgt_system" value="1"/>
    <arg name="tgt_component" value="1"/>
    <arg name="pluginlists_yaml" value="$(find mavprox_bridge)/config/mavros_distance_config.yaml"/>
    <arg name="config_yaml" value="$(find mavros)/launch/apm_config.yaml"/>
  </include>

  <!-- Camera-LiDAR Fusion System -->
  <include file="$(find camera_lidar_fusion)/launch/enhanced_system_jetson.launch"/>

  <!-- PointCloud to LaserScan Converter -->
  <node pkg="mavprox_bridge" type="pointcloud_to_laserscan_simple" name="pointcloud_to_laserscan_simple" output="screen">
    <param name="target_frame" value="base_link"/>
    <param name="min_height" value="-0.5"/>
    <param name="max_height" value="2.0"/>
    <param name="angle_min" value="-3.14159"/>
    <param name="angle_max" value="3.14159"/>
    <param name="angle_increment" value="0.0174533"/>
    <param name="range_min" value="0.1"/>
    <param name="range_max" value="50.0"/>
    <param name="scan_time" value="0.1"/>
  </node>

  <!-- Scan to Distance Sensor Bridge -->
  <node pkg="mavprox_bridge" type="scan_to_rangefinder" name="scan_to_rangefinder" output="screen">
    <param name="num_bins" value="72"/>
    <param name="min_distance" value="0.1"/>
    <param name="max_distance" value="50.0"/>
    <param name="angle_increment" value="5.0"/>
    <param name="angle_offset" value="0.0"/>
    <!-- Filtering parameters: only publish readings between 1.0-5.0 meters -->
    <param name="min_range" value="1.0"/>
    <param name="max_range" value="5.0"/>
  </node>

</launch>
